pkg/build/builder/cmd/builder.go

```diff
@@ -35,6 +36,7 @@ type builderConfig struct {
 	dockerClient    *docker.Client
 	dockerEndpoint  string
 	buildsClient    client.BuildInterface
+	eventsClient    kclient.EventInterface
```

Comment:
I'm not sure using events here is the right approach. Events, by design, are not reliable enough for those operation. I see you are protected against it with using timeouts, but isn't there a better option? unless I missed some discussion then you can forget what I've just said ;)